language-plutus-core/generators/Language/PlutusCore/Generators/Interesting.hs

```diff
@@ -6,39 +6,40 @@
 module Language.PlutusCore.Generators.Interesting
     ( TermGen
     , genOverapplication
-    , getBuiltinFactorial
+    , factorial
     , applyFactorial
     , genFactorial
     , genNaiveFib
     , genNatRoundtrip
-    , getBuiltinNatSum
+    , natSum
     , genListSum
     , genIfIntegers
     , fromInterestingTermGens
     ) where
 
-import           Language.PlutusCore
 import           Language.PlutusCore.Constant
 import           Language.PlutusCore.MkPlc
+import           Language.PlutusCore.Name
+import           Language.PlutusCore.Quote
+import           Language.PlutusCore.Type
 
 import           Language.PlutusCore.StdLib.Data.Bool
-import           Language.PlutusCore.StdLib.Data.Function
-import           Language.PlutusCore.StdLib.Data.List
+import           Language.PlutusCore.StdLib.Data.Function as Function
+import           Language.PlutusCore.StdLib.Data.List     as List
 import           Language.PlutusCore.StdLib.Data.Nat
 import           Language.PlutusCore.StdLib.Data.Unit
 import           Language.PlutusCore.StdLib.Meta
 import           Language.PlutusCore.StdLib.Type
 
 import           Language.PlutusCore.Generators
 
-import           Control.Monad.Morph
 import           Data.List                                (genericIndex)
 import           Hedgehog                                 hiding (Size, Var)
 import qualified Hedgehog.Gen                             as Gen
 import qualified Hedgehog.Range                           as Range
 
 -- | The type of terms-and-their-values generators.
-type TermGen size a = GenT Quote (TermOf (TypedBuiltinValue size a))
+type TermGen size a = Gen (TermOf (TypedBuiltinValue size a))
```

Comment:
Whenever we cross language boundaries, i.e.

1. generate a PLC term that has a Haskell counterpart
2. make a PLC term out of a Haskell value
3. make a Haskell value out of a PLC term

we can safely ignore `Quote`, because everything must be closed.